src/scripts/lib/autocomplete.js

```diff
@@ -31,12 +32,17 @@ const AutoComplete = function (el, item, elem) {
 AutoComplete.prototype.addEvents = function () {
   const that = this;
 
+  let dropdownRaceTimer;
   that.placeholderItem.addEventListener('click', function (e) {
-    setTimeout(function () {
+    clearTimeout(dropdownRaceTimer);
+
+    dropdownRaceTimer = setTimeout(function () {
       that.toggleDropdown();
-    }, 50);
+    }, 150);
```

Comment:
Hmm this delay feels quite unpleasant. Do you think it can be slightly lower or maybe there's another way?